lib/cc/cli/xdg_file.rb

```diff
@@ -0,0 +1,59 @@
+require "fileutils"
+require "yaml"
+
+module CC
+  module CLI
+    class XDGFile
```

Comment:
This class (and some of the others) seem to be mixing two very distinct concerns:

- Where is the file (XDG, etc)?
- How do I read/write a YAML file (which I'm surprised we have to deal with so directly)

I had recommended in another comment not worrying about XDG or namespaces _inside_ the container. We can just look for `/config.yml` and `/cache.yml`, defined as hard-coded constants (similar to how we handle `/code`). If you did that, I think a ton of this code goes away.